### ISSUE TITLE
The latest bridge library update causes crash on startup

### ISSUE THREAD

**Issue by lukebarnard1** [Half-Shot/matrix-appservice-twitter#59]
The matrix-org/matrix-appservice-bridge library was recently updated to use `0.7.3` of matrix-org/matrix-js-sdk. The twitter bridge depends on js-sdk version `0.5.3`, which does not have a particular function in the client instance, giving a stack trace:
```
Unhandled rejection TypeError: this._sdk.wrapRequest is not a function
    at ClientFactory.setLogFunction (/home/lukeb/dev/mx-as-twitter/node_modules/matrix-appservice-bridge/lib/components/client-factory.js:40:15)
    at Bridge.run (/home/lukeb/dev/mx-as-twitter/node_modules/matrix-appservice-bridge/lib/bridge.js:199:25)
```

The bridge should probably be using one version of the js-sdk: the same as the bridge library. Otherwise there would have to be two js-sdk dependencies installed somehow.

**Comment by lukebarnard1** [comment]
I saved this for a rainy day, and now feels relevant! It's a report that helped me upgrade the IRC bridge js-sdk dependency. The **TL;DR is that it's probably fine, provided that the twitter bridge only uses a subset of the following:**

### matrix-as-bridge API usage of matrix-js-sdk
comparing js-sdk versions 0.5.3 and 0.7.1

Example command to gather API changes: 
```
cd matrix-js-sdk
git diff v0.5.3..v0.7.1 -- lib/ | grep -A 20 -B 20 .prototype.sendStateEvent
```

client.credentials.userId
 - Seems to still exist, and still could be null

client._http.authedRequestWithPrefix()
 - Signature is the same
 - The diff looks like a refactor:
```diff
-        var fullUri = this.opts.baseUrl + prefix + path;
-        if (!queryParams) {
-            queryParams = {};
-        }
-        queryParams.access_token = this.opts.accessToken;
-        return this._request(
-            callback, method, fullUri, queryParams, data, localTimeoutMs
+        return this.authedRequest(
+            callback, method, path, queryParams, data, {
+                localTimeoutMs: localTimeoutMs,
+                prefix: prefix,
+            }
```
(and upon investigation, it was just a refactor that behaves very similarly if not identically)

client._http.opts._reqId
 - Never existed, and so was probably always ignored

client.sendEvent
 - E2E message sending has been factored out to a private function

client.createRoom
 - (SyncAPI.prototype.createRoom _was_ changed, but this is unimportant)

client.sendStateEvent
client.roomState
client.createRoom
client.getProfileInfo
client.createAlias
client.getStateEvent
 - Factored out to base-apis, but otherwise unchanged

client.sendTyping
client.sendReadReceipt
client.setPowerLevel
client.invite
client.kick
client.ban
client.unban
client.leave
client.setDisplayName
client.setAvatarUrl
client.joinRoom
 - No change
```

**Comment by Half-Shot** [comment]
Fixed by 66a65ba93c08ff103e6e1c02b559f887ac30b077
